### PR TITLE
docs: remove v3 to v4 migration docs

### PR DESCRIPTION
**Pull request by huangkairan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    docs: remove v3 to v4 migration docs       |
| 🇨🇳 Chinese |    移除 v3 至 v4 的迁移文档       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/breadcrumb/index.en-US.md [hunk 1]
```diff
@@ -15,36 +15,6 @@ A breadcrumb displays the current location within a hierarchy. It allows going b
...
-    <Menu.Item>item 2</Menu.Item>
-  </Menu>
-);
-return (
-  <Breadcrumb>
-    <Breadcrumb.Item overlay={menu}>Ant Design</Breadcrumb.Item>
-  </Breadcrumb>
-);
-```
-
```
Comment (MadCcc) [inline components/breadcrumb/index.en-US.md:47]: 这个感觉可以再留一段时间，还没有废弃多久

Comment (MadCcc) [reply components/breadcrumb/index.en-US.md:47]: 数据驱动的应该都可以留着，现在应该大部分用户都还是旧的写法

Comment (huangkairan) [reply components/breadcrumb/index.en-US.md:47]: > 这个感觉可以再留一段时间，还没有废弃多久

我感觉在 v5 里看到这些还挺奇怪的，就像我们不会在 v4 的文档中看到 v2 到 v3 的说明一样

Comment (huangkairan) [reply components/breadcrumb/index.en-US.md:47]: > 数据驱动的应该都可以留着，现在应该大部分用户都还是旧的写法
要不然再讨论下？之前私聊 @zombieJ 是说要移除掉

Comment (zombieJ) [reply components/breadcrumb/index.en-US.md:47]: v4 里有留档的，升级上来都会是新的。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**